plugins/chromatic_correction.py

```diff
@@ -0,0 +1,252 @@
+                sx = numpy.linalg.norm(mat[:2, 0])
+                sy = numpy.linalg.norm(mat[:2, 1])
+                scale2 = numpy.array([sx, sy])
+
+                # Compute rotation
+                rotation = numpy.arctan2(mat[1, 0] / sx, mat[0, 0] / sx)
+
+                # compute scale
+                scale = (1 / scale2[0], 1 / scale2[1])
+
+                # compute shear
+                a, b = mat[0, 0], mat[0, 1]
+                c, d = mat[1, 0], mat[1, 1]
+                shear = numpy.arctan2(a * c + b * d, scale[0] * scale[1])
+
+                # compute translation
+                tx = mat[0, 2]
+                ty = mat[1, 2]
+                pixel_size = im.metadata[model.MD_PIXEL_SIZE]
+                translation = (tx * pixel_size[0], -ty * pixel_size[1])
+
+                # stream settings
+                input_wl = im.metadata[model.MD_IN_WL]
+                output_wl = im.metadata[model.MD_OUT_WL]
+
+                try:
+                    fl_band = im.metadata[model.MD_EXTRA_SETTINGS]['Filter Wheel']['position'][0]['band']
+                    # The below metadata dictionary will be added to the microscope file using input command
+                    # under the filter wheel component, only if the metadata has Pixel size cor, Centre position cor,
+                    # Rotation cor, and Shear cor for a minimum one filter wheel band position
+                    trans_per_channel[model.MD_CHROMATIC_COR][fl_band] = {
+                        "Pixel size cor": [float(scale[0]), float(scale[1])],
+                        "Centre position cor": [float(translation[0]), float(translation[1])],
+                        "Rotation cor": float(rotation),
+                        "Shear cor": float(shear),
+                        # Extra information saved for tracking/debugging purpose
+                        "tint": list(im.metadata[model.MD_USER_TINT]),
+                        "excitation": [float(input_wl[0]), float(input_wl[1])],
+                        "emission": [float(output_wl[0]), float(output_wl[1])],
+                        "exposure time": float(im.metadata[model.MD_EXP_TIME]),
+                        "light power": float(im.metadata[model.MD_LIGHT_POWER])
+                    }
+
+                    # update metadata
+                    im.metadata.update({model.MD_PIXEL_SIZE_COR: scale})
+                    im.metadata.update({model.MD_POS_COR: translation})
+                    im.metadata.update({model.MD_ROTATION_COR: rotation})
+                    im.metadata.update({model.MD_SHEAR_COR: shear})
+
+                    # save the image and display
+                    self.show_static_stream(f"modified_{im.metadata[model.MD_DESCRIPTION]}_{i}.ome.tiff", im)
+                except (ValueError, KeyError) as ex:
+                    logging.error("Failed to compute chromatic aberration correction yaml file due to: %s", ex)
+
+        return trans_per_channel
```

Comment:
Can you log `trans_per_channel`, as logging.debug(). This way, if something goes wrong and the file is updated/overridden, or sometimes the value is incorrectly computed, it's easy to check the different values computed.